Mirror portraits setting for characters and the Join event (updated)

Jowan-Spooner opened the pull request:
This is a simple implementation adding a setting for each character. If enabled the portraits will be mirrored when they are shown on the right or the center-right positions. 

There was a discussion about how to implement this, so if you don't like it, don't merge it.

It should fix #168

coppolaemilio commented:
So, my feedback would be this:

1. Add the mirror option to the `Character Join` event:
![image](https://user-images.githubusercontent.com/2206700/114472047-de10d200-9bf1-11eb-8800-ad72c7e0bc7e.png)
_Super high fidelity mockup_ (Use `get_icon("MirrorX", "EditorIcons")` for the mirror icon)

2. Remove the "on right positions" and move it to where the offset is. This might be because the asset added is not looking to the proper side by default so leaving the option to mirror always will be handy. 
![image](https://user-images.githubusercontent.com/2206700/114472275-38aa2e00-9bf2-11eb-88b6-6a6b0a84e0ac.png)

So the portrait would check if to flip based on the character data and then again if to flip based on the `Character Join` event. That way regardless of how your sprites are set up, you have full control of where they are looking at. If you want all your sprites to be looking left but your artists sent you some character's sprites looking the other way you can flip them. And then when you add them on your timeline you can choose to flip them again at any time with the `Character Join` event.

Jowan-Spooner commented:
Okay I updated this and tested it on a little project made in 1.0. 

Here is the setting on the join event:
![Join Event](https://user-images.githubusercontent.com/42868150/114538908-2fea4400-9c54-11eb-9355-231b9dd1af22.png)

Here is the setting in the character editor:
![Character Editor](https://user-images.githubusercontent.com/42868150/114539009-4bede580-9c54-11eb-8171-6e08c8b1bc52.png)

I also updated the readme.

coppolaemilio commented:
amazing changes! thanks a lot 🙇🏻
